Cabal/Distribution/Simple/GHC.hs

```diff
@@ -53,6 +53,12 @@ module Distribution.Simple.GHC (
         pkgRoot
  ) where
 
+import Control.Applicative -- needed in 7.8, and not a warning in 7.10
```

Comment:
I'd prefer a shorter comment (e.g. "7.10 -Werror workaround" and maybe a link to https://ghc.haskell.org/trac/ghc/wiki/Migration/7.10#GHCsaysTheimportof...isredundant), since there are quite a few places in the codebase this change could be applied. Explaining it in detail everywhere seems excessive.